spec/spec.md

```diff
@@ -264,12 +265,15 @@ After [session establishment](#session-establishment) completes, the wallet endp
 
 After the [session establishment](#session-establishment) process completes, every message received by an endpoint is expected to be encrypted with AES-128-GCM (as specified by [NIST SP 800-38D](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf)). The sending endpoint should prepare the encrypted message by concatenating:
 
+- the message sequence number, a 4-byte big-endian unsigned integer
 - a random 12-byte IV (which should be newly generated for each encrypted message)
-- the AES-128-GCM message ciphertext, encrypted using the private key created during [session establishment](#session-establishment) and the random IV
+- the AES-128-GCM message ciphertext, encrypted using the private key created during [session establishment](#session-establishment) and the random IV, and including the 4-byte message sequence number as additional authorized data (AAD)
```

Comment:
Presumably I can supply this AAD as part of this SubtleCrypto API for the JS implementation?

```
interface AesGcmParams extends Algorithm {
    additionalData?: BufferSource; // <----
    iv: BufferSource;
    tagLength?: number;
}
```

Reply:
Yep! There's a draft PR up for this with partial JS implementation here: https://github.com/solana-mobile/mobile-wallet-adapter/pull/112